lib/src/main/java/com/dynatrace/file/util/WatchServiceBasedFilePoller.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2022 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.file.util;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
```

Comment:
Probably not relevant as it is not an LTS version, but this apparently breaks in Java 9 (`com.sun.*` packages do not have a backwards compatibility guarantee) :thinking:
https://stackoverflow.com/questions/46590600/how-can-i-specify-a-filewatcher-modifier-in-java9

But I think there's not really a solution for this short of implementing it on your own, so this comment is non-blocking for me. Maybe we could add a comment to it in case it comes up in the future? :thinking:

Reply:
I removed it again. It was part of the testing, and apparently some people managed to get it running smoothly on macOS by setting it. We did not have it before, so I assume we do not need it for Linux/Windows OSes.